Flaky WebSpa Orders test due to retrieve empty orders list

In some executions (it's not frequent at all), due to final race/timeouts, the OrderTest of WebSpa fails due to retrieving an empty order list and trying to get the last element (allocated in the numelements -1 position) with an array out of bounds exception. The Maven execution logs report:
```log
14:03:24  [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 42.98 s <<< FAILURE! -- in giis.eshopcontainers.e2e.functional.tests.webspa.WebSPAOrderTests
14:03:24  [ERROR] giis.eshopcontainers.e2e.functional.tests.webspa.WebSPAOrderTests.testCancelOrderSPA -- Time elapsed: 26.88 s <<< ERROR!
14:03:24  java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
14:03:24  	at java.base/jdk.internal.util.Preconditions.outOfBounds(Unknown Source)
14:03:24  	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Unknown Source)
14:03:24  	at java.base/jdk.internal.util.Preconditions.checkIndex(Unknown Source)
14:03:24  	at java.base/java.util.Objects.checkIndex(Unknown Source)
14:03:24  	at java.base/java.util.ArrayList.get(Unknown Source)
14:03:24  	at giis.eshopcontainers.e2e.functional.utils.OrdersWebSPA.cancelLastOrder(OrdersWebSPA.java:141)
14:03:24  	at giis.eshopcontainers.e2e.functional.tests.webspa.WebSPAOrderTests.testCancelOrderSPA(WebSPAOrderTests.java:66)
14:03:24  
14:03:24  [INFO] 
14:03:24  [INFO] Results:
14:03:24  [INFO] 
14:03:24  [ERROR] Errors: 
14:03:24  [ERROR]   WebSPAOrderTests.testCancelOrderSPA:66 » IndexOutOfBounds Index -1 out of bounds for length 0
14:03:24  [INFO] 
14:03:24  [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0

```
This should be solved/mitigated implementing a retrieval/retry mechanism that avoids this type of problems-failures

--- a/src/test/java/giis/eshopcontainers/e2e/functional/utils/Orders.java
+++ b/src/test/java/giis/eshopcontainers/e2e/functional/utils/Orders.java
@@ -125,12 +125,15 @@ public class Orders extends Shopping {
     /**
      * Navigates to the orders menu and press the Cancellation button of the last order. Checks that the order state
      * evolves as expected.
+     * Waits for the orders list to be non-empty before accessing it to guard against transient empty-list responses.
      *
      * @param driver {@code WebDriver} on which the operations are performed.
      * @param waiter {@code Waiter} to perform the necessary async waits.
      */
     public void cancelLastOrder(WebDriver driver, Waiter waiter) throws ElementNotFoundException {
         navUtils.toOrdersPage(driver, waiter);
+        waiter.waitUntil(ExpectedConditions.numberOfElementsToBeMoreThan(By.className("esh-orders-items"), 0),
+                "Orders list did not appear after navigation to orders page");
         List<WebElement> listOrders = driver.findElements(By.className("esh-orders-items"));
         WebElement lastOrder = listOrders.get(listOrders.size() - 1);
         WebElement cancelButton = lastOrder.findElement(By.linkText("Cancel"));

--- a/src/test/java/giis/eshopcontainers/e2e/functional/utils/OrdersWebSPA.java
+++ b/src/test/java/giis/eshopcontainers/e2e/functional/utils/OrdersWebSPA.java
@@ -130,6 +130,8 @@ public class OrdersWebSPA extends Orders {
 
     /**
      * Cancels the last order in the WebSPA frontend (SPA version).
+     * Waits for the orders list to be non-empty before accessing it: the SPA fetches
+     * orders asynchronously so the list can be transiently empty after navigation.
      *
      * @param driver {@code WebDriver} on which the operations are performed.
      * @param waiter {@code Waiter} to perform the necessary async waits.
@@ -137,6 +139,8 @@ public class OrdersWebSPA extends Orders {
     @Override
     public void cancelLastOrder(WebDriver driver, Waiter waiter) throws ElementNotFoundException {
         navUtils.toOrdersPage(driver, waiter);
+        waiter.waitUntil(ExpectedConditions.numberOfElementsToBeMoreThan(By.className("esh-orders-item"), 0),
+                "Orders list did not appear after navigation to orders page");
         List<WebElement> listOrders = driver.findElements(By.className("esh-orders-item"));
         WebElement lastOrder = listOrders.get(listOrders.size() - 1);
         WebElement cancelLink = lastOrder.findElement(By.linkText("Cancel"));

